### PR TITLE
Add side-scan sonar to SAM simulation

### PR DESCRIPTION
**Pull request by luxiya01**

### CODE REVIEW

**Comment by nilsbore** [comment]
@luxiya01 Looks good! If you could also add a conversion to the [smarc_msgs sidescan format](https://github.com/smarc-project/smarc_msgs/blob/noetic-devel/msg/Sidescan.msg) that would be great!

**Comment by nilsbore** [comment]
Preferably in the [bridge node](https://github.com/smarc-project/smarc_stonefish_sims/blob/noetic-devel/sam_stonefish_sim/src/sam_sim_msg_bridge.cpp)

File: sam_stonefish_sim/data/sam_auv/sam_auv.xml [hunk 1]
```diff
@@ -256,6 +256,16 @@
...
     </sensor>
 
+    <sensor name="side_scan_sonar" type="sss" rate="5.0">
+        <link name="Vehicle"/>
+        <origin rpy="0.0 3.1416 1.5707" xyz="0.7 0.0 -0.07"/>
+        <specs bins="500" lines="400" horizontal_beam_width="0.5" vertical_beam_width="60.0" vertical_tilt="60.0"/>
+        <settings range_min="1.0" range_max="100.0" gain="1.0"/>
+        <noise multiplicative="0.02" additive="0.04"/>
+        <display colormap="hot"/>
+        <ros_publisher topic="/$(arg robot_name)/perception/sss"/>
```
Comment (nilsbore) [inline sam_stonefish_sim/data/sam_auv/sam_auv.xml]: Let's call this `/$(arg robot_name)/sim/sidescan` instead, then we'll add a conversion to the actual sidescan topic

Comment (nilsbore) [reply sam_stonefish_sim/data/sam_auv/sam_auv.xml]: Which should be named `/$(arg robot_name)/payload/sidescan` according to [the spec](https://github.com/smarc-project/smarc_msgs#payloads)

**Comment by luxiya01** [comment]
@nilsbore The bridge node is updated with conversions to smarc_msg sss type as requested :)
See also https://github.com/smarc-project/smarc_utils/pull/33 that contains small modifications to the view_sss node

**Comment by nilsbore** [comment]
Cool! I was also running the view_sss node today so I probably did the same changes you did :)

**Comment by nilsbore** [comment]
Changes look great, merging.

**Comment by nilsbore** [comment]
I added https://github.com/smarc-project/smarc_stonefish_sims/issues/5 for completing the rest of the sidescan interface, it's not urgent.

**Pull request merged**